app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,101 @@
+class PasswordResetsController < ApplicationController
+  def new
+  end
+
+  def create
+    if params[:username].blank?
+      flash.now[:alert] = 'Please enter a username'
+      render 'new', status: :unprocessable_entity
+      return
+    end
+
+    begin
+      user = User.find(escaped_username)
+
+      if user.nil?
+        user_not_found
+      else
+        PasswordResetMailer.password_reset(user).deliver
+        flash.now[:notice] = 'Your password reset email has been sent'
+        @status = :ok
+      end
+    rescue Net::HTTPServerException => e
+      if e.response.code.to_i == 404
+        user_not_found
+      elsif e.response.code.to_i == 406
+        flash.now[:alert] = "'#{params[:username]}' is not a valid username"
+        @status = :forbidden
+      else
+        raise
+      end
+    end
+
+    render 'new', status: @status
```

Comment:
While in the current code, the @status (hope someone doesn't have that as a github name) will be set in every case, it might be worth setting it to a default value as soon as the method is entered, to ensure we always return something valid. This protects agains future code changes that might introduce new branches that don't set it.

It's perhaps more a stylistic approach, but I think one worth considering.